From 94a268a68172e76675bbf67d186e57a0f20c4a4a Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 10 Jun 2025 19:25:25 +0200
Subject: [PATCH] tee: optee: move SMC ABI macros definition

Move definition of macros OPTEE_SCM_FUNCID_CALL_WITH_{RPC|REGD}_ARG
to better reflect the function IDs order.

Change-Id: I5ed5a58302afc1adc3ead0678c51b74fbc47a905
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/466079
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/tee/optee/optee_smc.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/tee/optee/optee_smc.h b/drivers/tee/optee/optee_smc.h
index b0972f630efb..bf4a92002d9f 100644
--- a/drivers/tee/optee/optee_smc.h
+++ b/drivers/tee/optee/optee_smc.h
@@ -449,6 +449,12 @@ struct optee_smc_disable_shm_cache_result {
 #define OPTEE_SMC_IT_NOTIF_VALUE_VALID		BIT(0)
 #define OPTEE_SMC_IT_NOTIF_VALUE_PENDING	BIT(1)
 
+/* See OPTEE_SMC_CALL_WITH_RPC_ARG above */
+#define OPTEE_SMC_FUNCID_CALL_WITH_RPC_ARG	18
+
+/* See OPTEE_SMC_CALL_WITH_REGD_ARG above */
+#define OPTEE_SMC_FUNCID_CALL_WITH_REGD_ARG	19
+
 /*
  * Notification that OP-TEE generates and interruption.
  */
@@ -460,12 +466,6 @@ struct optee_smc_disable_shm_cache_result {
 #define OPTEE_SMC_SET_IT_NOTIF_MASK \
 	OPTEE_SMC_FAST_CALL_VAL(OPTEE_SMC_FUNCID_SET_IT_NOTIF_MASK)
 
-/* See OPTEE_SMC_CALL_WITH_RPC_ARG above */
-#define OPTEE_SMC_FUNCID_CALL_WITH_RPC_ARG	18
-
-/* See OPTEE_SMC_CALL_WITH_REGD_ARG above */
-#define OPTEE_SMC_FUNCID_CALL_WITH_REGD_ARG	19
-
 /*
  * Resume from RPC (for example after processing a foreign interrupt)
  *
-- 
2.53.0

